compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -779,7 +779,7 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(implicit ctx: Context) {
               newOwners = ctx.owner :: Nil,
               substFrom = ddef.vparamss.head.map(_.symbol),
               substTo = argSyms)
-            Inlined(ddef, bindingsBuf.toList, expander.transform(ddef.rhs))
+            Block(bindingsBuf.toList, expander.transform(ddef.rhs))
```

Comment:
We missed something up in [line 767](https://github.com/lampepfl/dotty/pull/7365/files#diff-021cdae2fbad688e5717101b37e981d1R767). We calls `closureDef` which strips `Inlined` nodes the we should keep.